Source/spells.cpp

```diff
@@ -246,14 +246,15 @@ void CastSpell(int id, int spl, int sx, int sy, int dx, int dy, int spllvl)
  * @param pnum player index
  * @param rid target player index
  */
-void DoResurrect(int pnum, int rid)
+void DoResurrect(int pnum, uint16_t rid)
 {
 	if (pnum == MyPlayerId) {
 		NewCursor(CURSOR_HAND);
 	}
 
-	if (rid == -1)
+	if ((DWORD)pnum >= MAX_PLRS || rid >= MAX_PLRS) {

```

Comment:
Why would either of these ever be greater than `MAX_PLRS`?

Can't we make `rid` `optional<int>` instead?

Reply:
Because they're parameters sent via network messages so they need to be validated regardless.

Reply:
We have added a bit of validation for these values, but the more the better before we remove the encryption requirement and implement a game list :hear_no_evil:

Reply:
That being said `pnum` should have been validated before this point or we would have OOBs in `ParseCmd()`

Reply:
FYI, this was previously discussed in #2222.

Reply:
I just copy/pasted the code from heal other, if you think it's wrong, fix heal other too ;)